File: Examples/CSharp/AsposePDF/Forms/DetermineRequiredField.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using System;

namespace CSharp.AsposePdf.Forms
{
    public class DetermineRequiredField
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Forms();

            // Load source PDF file
            Document pdf = new Document(dataDir + "DetermineRequiredField.pdf");

            // Instantiate Form object
            Aspose.Pdf.Facades.Form pdfForm = new Aspose.Pdf.Facades.Form(pdf);

            // Iterate through each field inside PDF form
            foreach (Aspose.Pdf.InteractiveFeatures.Forms.Field field in pdf.Form.Fields)
            {
                // Determine if the field is marked as required or not
                bool isRequired = pdfForm.IsRequiredField(field.FullName);
                if (isRequired)

                {
                    // Print either the field is marked as required or not
                    Console.WriteLine("The field named " + field.FullName + " is required");
                }
            }

        }
    }
}

File: Examples/VisualBasic/AsposePDF/Forms/DetermineRequiredField.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf
Imports APF = Aspose.Pdf.Facades
Imports Aspose.Pdf.Forms

Namespace AsposePDF.Forms
    Public Class DetermineRequiredField
        Public Shared Sub Run()
            ' ExStart:DetermineRequiredField
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Forms()

            ' Load source PDF file
            Dim pdf As New Document(dataDir & Convert.ToString("DetermineRequiredField.pdf"))

            ' Instantiate Form object
            Dim pdfForm As New APF.Form(pdf)

            ' Iterate through each field inside PDF form
            For Each field As Field In pdf.Form.Fields
                ' Determine if the field is marked as required or not
                Dim isRequired As Boolean = pdfForm.IsRequiredField(field.FullName)
                If isRequired Then
                    ' Print either the field is marked as required or not
                    Console.WriteLine("The field named " + field.FullName + " is required")
                End If
            Next
            ' ExEnd:DetermineRequiredField
        End Sub
    End Class
End Namespace
